src/unum/util/util_common.h

```diff
@@ -317,4 +317,7 @@ void util_build_url(char *proto, int type, char *url, unsigned int length,
 // doesn't support this, it should return without doing anything.
 int __attribute__((weak)) platform_release_renew(void);
 
+// Check if the interface is a guest network
```

Comment:
This doesn't look right. Judging by its name util_is_guest_network() is a common code function, so it shouldn't be declared weak. 
I think we need to rethink how to make this work. The information about "type" of the interface (guest or not) is already available in the util_enum_ifs(), we just need to find a way to propagate it to the caller in a way that's not breaking the compatibility with the old code (maybe extend the original and turn it into util_enum_ifs_ext() function, but keep providing util_enum_ifs() wrapper for compatibility purposes). 
Please create an issue for me to extend the util_enum_ifs() with more information about the interfaces it reports.

Reply:
Done